### ISSUE TITLE
String index out of range: -5

### ISSUE THREAD

**Issue by MiguelCatalan** [pedrovgs/AndroidWiFiADB#2]
From a Windows environment I get this Exception:

``` clojure
String index out of range: -5
java.lang.StringIndexOutOfBoundsException: String index out of range: -5
    at java.lang.String.substring(String.java:1954)
    at com.github.pedrovgs.androidwifiadb.adb.ADBParser.parseGetDeviceIp(ADBParser.java:69)
    at com.github.pedrovgs.androidwifiadb.adb.ADB.getDeviceIp(ADB.java:68)
    at com.github.pedrovgs.androidwifiadb.adb.ADB.connectDeviceByIp(ADB.java:56)
    at com.github.pedrovgs.androidwifiadb.adb.ADB.connectDevices(ADB.java:49)
    at com.github.pedrovgs.androidwifiadb.AndroidWiFiADB.connectDevices(AndroidWiFiADB.java:43)
    at com.github.pedrovgs.androidwifiadb.action.AndroidWiFiADBAction$1.run(AndroidWiFiADBAction.java:51)
    at com.intellij.openapi.application.impl.ApplicationImpl$8.run(ApplicationImpl.java:400)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
    at org.jetbrains.ide.PooledThreadExecutor$1$1.run(PooledThreadExecutor.java:56)
```

**EDIT:**
FYI it is a Windows 10

**Comment by pedrovgs** [comment]
@MiguelCatalan can you paste here the output obtained executing the command `adb shell ip -f inet addr show wlan0` in your terminal?

**Comment by MiguelCatalan** [comment]
Sure:

``` c
22: wlan0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP qlen 1000
inet 192.168.1.134/24 brd 192.168.1.255 scope global wlan0
```

**Comment by mdogancay** [comment]
@pedrovgs please change your code and rebuild your project:
(com/github/pedrovgs/androidwifiadb/adb/ADBParser.java)

``` Java
  public String parseGetDeviceIp(String ipInfo) {

    if (ipInfo.isEmpty() || ipInfo.contains(ERROR_PARSING_DEVICE_IP_KEY)) {
      return "";
    }
    String[] splittedOutput = ipInfo.split("\\n");
    int line;
    for ( line=0; line < splittedOutput.length; line++ ) {
      if ( splittedOutput[line].contains(START_DEVICE_IP_INDICATOR) ) {
        break;
      }
    }
    int end = splittedOutput[line].indexOf(END_DEVICE_IP_INDICATOR);
    int start = splittedOutput[line].indexOf(START_DEVICE_IP_INDICATOR) + 5;
    return splittedOutput[line].substring(start, end);
  }
```

This will add a basic search mechanism for the true line number.
I will wait next build.

**Comment by pedrovgs** [comment]
@MiguelCatalan @mdogancay can you check if this version is working? https://drive.google.com/file/d/0B9xkpTnF9BXjSk9VeU92MUt5M1E/view?usp=sharing Remember to uninstall the current version and restart the IDE before to install this one. Thanks!

**Comment by mdogancay** [comment]
Hi @MiguelCatalan , I tested it now and it's works for me.
Cheers!

**Comment by pedrovgs** [comment]
Fixed in the Release 1.1 I've just published :)
